src/main/java/uk/gov/register/presentation/resource/ClientErrorExceptionMapper.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.register.presentation.resource;
+
+import uk.gov.register.presentation.representations.ExtraMediaType;
+import uk.gov.register.presentation.view.ViewFactory;
+import uk.gov.register.thymeleaf.ThymeleafView;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ClientErrorException;
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+public class ClientErrorExceptionMapper implements ExceptionMapper<ClientErrorException> {
+
+    private ViewFactory viewFactory;
+
+    @Inject
+    public ClientErrorExceptionMapper(ViewFactory viewFactory) {
+        this.viewFactory = viewFactory;
+    }
+
+    @Override
+    public Response toResponse(ClientErrorException exception) {
+        return Response
+                .status(exception.getResponse().getStatus())
+                .header(HttpHeaders.CONTENT_TYPE, ExtraMediaType.TEXT_HTML)
+                .entity(getEntityFor(exception))
+                .build();
+    }
+
+    private ThymeleafView getEntityFor(ClientErrorException exception) {
+        if (exception instanceof BadRequestException) {
+            return viewFactory.badRequestExceptionView((BadRequestException) exception);
+        }
+
+        if (exception instanceof NotFoundException || exception instanceof NotAcceptableException) {
+            return viewFactory.thymeleafView("404.html");
```

Comment:
The 406 response is returned but html page displays "the page is not found" so reusing the `404.html` template. In case we want some other display text we can create new template. 
If it is confusing we can rename the template file.